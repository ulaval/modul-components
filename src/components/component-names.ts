export const ACCORDION_GROUP_NAME: string = 'm-accordion-group';
export const ACCORDION_NAME: string = 'm-accordion';
export const LIST_ITEM_NAME: string = 'm-list-item';
export const BUTTON_NAME: string = 'm-button';
export const BUTTON_GROUP_NAME: string = 'm-button-group';
export const CAROUSEL_NAME: string = 'm-carousel';
export const CAROUSEL_ITEM_NAME: string = 'm-carousel-item';
export const CHECKBOX_NAME: string = 'm-checkbox';
export const DATEFIELDS_NAME: string = 'm-datefields';
export const DATEPICKER_NAME: string = 'm-datepicker';
export const DIALOG_NAME: string = 'm-dialog';
export const DROPDOWN_NAME: string = 'm-dropdown';
export const DROPDOWN_ITEM_NAME: string = 'm-dropdown-item';
export const DROPDOWN_GROUP_NAME: string = 'm-dropdown-group';
export const DYNAMIC_TEMPLATE_NAME: string = 'm-dynamic-template';
export const FILE_SELECT_NAME: string = 'm-file-select';
export const FILE_UPLOAD_NAME: string = 'm-file-upload';
export const EDIT_WINDOW_NAME: string = 'm-edit-window';
export const FLEX_TEMPLATE_NAME: string = 'm-flex-template';
export const I18N_NAME: string = 'm-i18n';
export const ICON_NAME: string = 'm-icon';
export const ICON_BUTTON_NAME: string = 'm-icon-button';
export const INPUT_STYLE_NAME: string = 'm-input-style';
export const LIMIT_TEXT_NAME: string = 'm-limit-text';
export const LINK_NAME: string = 'm-link';
export const MESSAGE_NAME: string = 'm-message';
export const MODAL_NAME: string = 'm-modal';
export const NAVBAR_NAME: string = 'm-navbar';
export const NAVBAR_ITEM_NAME: string = 'm-navbar-item';
export const MENU_NAME: string = 'm-menu';
export const MENU_ITEM_NAME: string = 'm-menu-item';
export const PANEL_NAME: string = 'm-panel';
export const PHONE_NUMBER_NAME: string = 'm-phone-number';
export const POPPER_NAME: string = 'm-popper';
export const POPUP_NAME: string = 'm-popup';
export const PROGRESS_NAME: string = 'm-progress';
export const RADIO_NAME: string = 'm-radio';
export const RADIO_GROUP_NAME: string = 'm-radio-group';
export const RADIO_STYLE_NAME: string = 'm-radio-style';
export const SCROLL_TOP_NAME: string = 'm-scroll-top';
export const SIDEBAR_NAME: string = 'm-sidebar';
export const SLIDER_NAME: string = 'm-slider';
export const SPINNER_NAME: string = 'm-spinner';
export const STATUS_NAME: string = 'm-status';
export const STEP_NAME: string = 'm-step';
export const STEPPERS_NAME: string = 'm-steppers';
export const STEPPERS_ITEM_NAME: string = 'm-steppers-item';
export const SWITCH_NAME: string = 'm-switch';
export const TAB_PANEL_NAME: string = 'm-tab-panel';
export const TABS_NAME: string = 'm-tabs';
export const TEMPLATE_NAME: string = 'm-template';
export const TEXTAREA_NAME: string = 'm-textarea';
export const TEXTFIELD_NAME: string = 'm-textfield';
export const TIMEPICKER_NAME: string = 'm-timepicker';
export const TOOLTIP_NAME: string = 'm-tooltip';
export const UPLOAD_NAME: string = 'm-upload';
export const UPLOAD_INPUT_NAME: string = 'm-upload-input';
export const UPLOAD_DRAGDROP_NAME: string = 'm-upload-dragdrop';
export const UPLOAD_FILESLIST_NAME: string = 'm-upload-fileslist';
export const VALIDATION_MESSAGE_NAME: string = 'm-validation-message';
