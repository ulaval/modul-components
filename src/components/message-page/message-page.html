<article class="m-message-page"
         :class="{'m--is-state-confirmation': isStateConfirmation,
                  'm--is-state-information': isStateInformation,
                  'm--is-state-warning': isStateWarning,
                  'm--is-state-error': isStateError,
                  'm--is-skin-default': isSkinDefault,
                  'm--is-skin-light': isSkinLight }">
    <header class="m-message-page__header">
        <h2 class="m-message-page__title"
            v-if="title"
            v-html="title"></h2>
        <div class="m-message-page__icon-container"
             aria-hidden="true">
            <div class="m-message-page__icon"
                :style="styleObject"
                v-html="svg"
                v-if="isSvg"></div>
            <m-icon class="m-message-page__icon"
                    :name="iconNameProp"
                    :size="propImageSize"
                    v-else></m-icon>
        </div>
    </header>
    <div class="m-message-page__body"
         v-if="hasBody">
        <p class="m-message-page__hints"
            :class="{'m--no-content': !hasLinksAndSlot}"
            v-if="hasHints"
            v-for="hint in hints"
            ref="hint">{{hint}}</p>
        <div class="m-message-page__details"
            v-if="$slots.default">
            <slot></slot>
        </div>
        <ul class="m-message-page__links"
            v-if="hasLinks">
            <li ref="link"
                v-for="link in links">
                <m-link class="m-message-page__link"
                        icon-name="m-svg__chevron--right"
                        mode="link" :url="link.url"
                        :target="isTargetExternal(link.external)">{{link.label}}</m-link>
            </li>
        </ul>
    </div>
</article>
