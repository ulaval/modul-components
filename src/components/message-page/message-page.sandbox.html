<div>
    <h5>default - different skins ans icons</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page svg-name="m-svg__rocket"
                        state="information"
                        title="Votre titre">
        </m-message-page>
    </div>

    <br />

    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="information" :title="title" icon-name="m-svg__file-video"></m-message-page>
    </div>

    <br />

    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="information" :title="title"></m-message-page>
    </div>

    <br />

    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="confirmation" :title="title"></m-message-page>
    </div>

    <br />

    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="warning" :title="title"></m-message-page>
    </div>

    <hr/>

    <h5>with #one hint</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="error" :title="title" icon-name="error"  :hints="oneHint"></m-message-page>
    </div>

    <hr/>

    <h5>with #multiple hints</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="error" :title="title" icon-name="error"  :hints="manyHints"></m-message-page>
    </div>

    <hr/>

    <h5>with #one link</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="error" :title="title" icon-name="error"  :links="oneLink"></m-message-page>
    </div>

    <hr/>

    <h5>with #multiple links</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="error" :title="title" icon-name="error"  :links="manyLinks"></m-message-page>
    </div>

    <hr/>

    <h5>unexpected error popup with hints and links</h5>
    <m-button v-m-popup:errorpopup>pop me an error</m-button>
    <m-modal ref="errorpopup" :negativeLink="false" ok-label="Fermer">
            <m-message-page state="error" :title="title" icon-name="m-svg__warning-filled" :hints="manyHints" :links="manyLinks"></m-message-page>
    </m-modal>

    <hr/>

    <h5>with content in slot</h5>
    <div style="border: solid 1px black; padding: 10px; width: 600px;">
        <m-message-page state="error" :title="title" icon-name="error" :hints="manyHints" :links="manyLinks">
            <span>
                <b>***Custom error details content through slot.***</b>
            </span>
        </m-message-page>
    </div>

    <hr/>

</div>
