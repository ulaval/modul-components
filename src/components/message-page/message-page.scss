@import '../../styles/abstracts/variables';

$m-message-page--shadow-width: 150px;
$m-message-page--shadow-height: 5px;

@mixin m-message-page-state($color, $color-background) {
    .m-message-page {
        &__title,
        &__icon {
            color: $color;
        }

        &__icon-container::before {
            background: $color-background;
        }
    }
}

.m-message-page {
    &.m--is-state-confirmation {
        @include m-message-page-state($m-color--success, $m-color--success-lightest);
    }

    &.m--is-state-information {
        @include m-message-page-state($m-color--interactive, $m-color--interactive-lightest);
    }

    &.m--is-state-error {
        @include m-message-page-state($m-color--error, $m-color--error-lightest);
    }

    &.m--is-state-warning {
        @include m-message-page-state($m-color--warning, $m-color--warning-lightest);
    }

    &.m--is-skin-default {
        .m-message-page {
            &__title {
                font-size: $m-font-size--h3;
            }

            &__icon-container {
                &::before {
                    right: 0;
                    left: 0;
                    transform: translate3d(0, -50%, 0);
                    height: 80px; // Magic number
                }

                &::after {
                    width: $m-message-page--shadow-width;
                }
            }
        }
    }

    &.m--is-skin-light {
        .m-message-page {
            &__title {
                font-size: $m-font-size--h4;
            }

            &__icon-container {
                &::before {
                    left: 50%;
                    transform: translate3d(-50%, -50%, 0);
                    width: 100%;
                    max-width: 140px; //Magic number
                    height: 48px; // Magic number
                }

                &::after {
                    width: $m-message-page--shadow-width/2;
                }
            }
        }
    }

    &__header {
        ~ .m-message-page__body {
            padding-top: $m-spacing;
        }
    }

    &__title {
        text-align: center;
        font-weight: $m-font-weight--black;
        margin: 0;

        ~ .m-message-page__icon-container {
            margin-top: $m-spacing;
        }
    }

    &__icon-container {
        position: relative;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;

        &::before {
            position: absolute;
            top: calc(50% - #{$m-spacing--s - $m-message-page--shadow-height}); // Magic number: align this backgroun with icon
            z-index: 0;
            content: '';
            display: block;
        }

        &::after {
            content: '';
            height: $m-message-page--shadow-height;
            border-radius: 50%;
            background: $m-color--grey-lighter;
            margin-top: $m-spacing--s;
        }

        &__svg {
            z-index: 1;
        }
    }

    &__icon {
        position: relative;
    }

    &__stacktrace {
        margin-top: $m-margin;
        max-height: 120px;
        overflow: auto;
        padding: $m-padding;

        @include m-scrollbar();

        pre {
            margin: 0;
        }
    }

    &__hints,
    &__links {
        text-align: left; // needed when inside a popup
    }

    &__hints {
        margin: 0;

        &.m--no-content {
            text-align: center;
        }

        + .m-message-page__hints {
            margin-top: $m-margin;
        }
    }

    &__details {
        margin: 0;

        &:not(:first-child) {
            margin: $m-spacing 0 0 0;
        }
    }

    &__links {
        list-style: none;
        margin: 0;
        padding: 0;

        &:not(:first-child) {
            margin: $m-spacing 0 0 0;
        }
    }
}
