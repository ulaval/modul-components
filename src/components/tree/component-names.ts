export const TREE_NODE_NAME: string = 'm-tree-node';
export const TREE_ICON_NAME: string = 'm-tree-icon';
