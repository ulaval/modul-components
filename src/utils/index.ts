import Vue, { PluginObject } from 'vue';

import MediaQueriesPlugin from './media-queries/media-queries';
import ModulPlugin from './modul/modul';
import MessagesPlugin from './i18n/i18n';
import HttpPlugin, { HttpPluginOptions } from './http/http';
import SpritesPlugin from './svg/sprites';
import * as TouchPlugin from 'vue-touch';
import ConfirmPlugin from './modal/confirm';
import AlertPlugin from './modal/alert';
import FilePlugin from '../utils/file/file';
import { PortalPluginInstall } from 'portal-vue';
import LoggerPlugin, { ConsoleOptions } from './logger/logger';

export interface UtilsPluginOptions {
    httpPluginOptions?: HttpPluginOptions;
    consoleOptions?: ConsoleOptions;
}

const UtilsPlugin: PluginObject<any> = {
    install(v, options): void {
        if (!v.prototype.$log) {
            Vue.use(LoggerPlugin, options ? options.consoleOptions : undefined);
        } else if (options) {
            v.prototype.$log.setConsoleOptions(options.consoleOptions);
        }

        Vue.use(MediaQueriesPlugin);
        Vue.use(ModulPlugin);
        Vue.use(MessagesPlugin);
        Vue.use(HttpPlugin, options ? options.httpPluginOptions : undefined);
        Vue.use({ install: PortalPluginInstall });
        Vue.use(SpritesPlugin);
        Vue.use(TouchPlugin, { name: 'v-touch' });
        Vue.use(ConfirmPlugin);
        Vue.use(AlertPlugin);
        Vue.use(FilePlugin);
    }
};

export default UtilsPlugin;
