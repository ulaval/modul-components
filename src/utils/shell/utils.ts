import Vue from 'vue';
import { Shell, PackageOptions } from '@ulaval/shell-ui/dist/shell/shell';

declare var __webpack_public_path__: string | undefined;

export type CreerInstanceVueFn = (shell: Shell, rootPath: string) => Vue;

export function setupShell(nomPackage: string, creerInstanceVue: CreerInstanceVueFn): void {
    window[nomPackage] = {
        mount(shell: Shell, options: PackageOptions): Promise<void> {
            if (this.vue) {
                console.warn('Already mounted.');
                return Promise.resolve();
            }

            __webpack_public_path__ = options.repoPublicPath;

            this.idElementRacine = options.rootElement;

            let rootPath: string = options.rootPath as string;
            if (rootPath[rootPath.length - 1] !== '/') {
                rootPath += '/';
            }

            this.vue = creerInstanceVue(shell, rootPath);

            this.vue.$mount('#' + this.idElementRacine);

            return Promise.resolve();
        },

        unmount(): Promise<void> {
            if (!this.vue) {
                return Promise.resolve();
            }

            const el = this.vue.$el;
            this.vue.$destroy();
            this.vue = undefined;

            let div = document.createElement('div');
            div.id = this.idElementRacine;
            if (el.parentNode) {
                el.parentNode.replaceChild(div, el);
            }

            return Promise.resolve();
        }
    };
}
