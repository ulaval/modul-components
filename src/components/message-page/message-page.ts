import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';

import { ModulVue } from '../../utils/vue/vue';
import AccordionPlugin from '../accordion/accordion';
import { MESSAGE_PAGE_NAME } from '../component-names';
import I18nPlugin from '../i18n/i18n';
import LinkPlugin from '../link/link';
import { MMessageState } from '../message/message';
import WithRender from './message-page.html?style=./message-page.scss';

/**
 * Utility class to manage the properties relatied to the link displayed in the error pages.
 */
export class Link {
    /**
     * Constructor
     * @param label Label for the url link to display.
     * @param url Target for the location to navigate to, can be relative.
     * @param external Defines to the target is external (opens in new tab), or internal (opens in same tab) to the application.
     */
    constructor(public label: string, public url: string, public external: boolean = false) {
    }
}

export enum MMessagePageSkin {
    Default = 'default',
    Light = 'light'
}

export enum MMessagePageImageSize {
    Default = '130px',
    Small = '76px'
}

@WithRender
@Component
export class MMessagePage extends ModulVue {

    @Prop({
        default: 'error',
        validator: value =>
            value === MMessageState.Information ||
            value === MMessageState.Warning ||
            value === MMessageState.Confirmation ||
            value === MMessageState.Error
    })
    public state: string;

    @Prop({
        default: MMessagePageSkin.Default,
        validator: value =>
            value === MMessagePageSkin.Default ||
            value === MMessagePageSkin.Light
    })
    public skin: string;

    @Prop()
    public iconName: string;

    @Prop()
    public svgName: string;

    @Prop()
    public imageSize: string;

    @Prop()
    public title: string;

    @Prop({ default: () => [] })
    public hints: string[];

    @Prop({ default: () => [] })
    public links: Link[];

    public svg: string;

    public get isSvg(): boolean {
        return (this.svgName !== undefined) && (this.svgName.trim().length > 0);
    }

    public get hasHints(): boolean {
        return this.hints.length > 0;
    }

    public get hasLinks(): boolean {
        return this.links.length > 0;
    }

    public get styleObject(): { [name: string ]: string } {
        return {
            width: this.propImageSize
        };
    }

    public get propImageSize(): string {
        if (this.imageSize) {
            return this.imageSize;
        }
        return this.isSkinLight ? MMessagePageImageSize.Small : MMessagePageImageSize.Default;
    }

    public isTargetExternal(isExternal: boolean): string {
        return isExternal ? '_blank' : '' ;
    }

    protected created(): void {
        if (this.svgName) {
            this.svg = require(`../../assets/icons/svg/${this.svgName}.svg`);
        }
    }

    private get hasLinksAndSlot(): boolean {
        return this.hasLinks || !!this.$slots.default;
    }

    private get hasBody(): boolean {
        return this.hasHints || this.hasLinks || !!this.$slots.default;
    }

    private get isSkinDefault(): boolean {
        return this.skin === MMessagePageSkin.Default;
    }

    private get isSkinLight(): boolean {
        return this.skin === MMessagePageSkin.Light;
    }

    private get isStateInformation(): boolean {
        return this.state === MMessageState.Information;
    }

    private get isStateWarning(): boolean {
        return this.state === MMessageState.Warning;
    }

    private get isStateError(): boolean {
        return this.state === MMessageState.Error;
    }

    private get isStateConfirmation(): boolean {
        return this.state === MMessageState.Confirmation;
    }

    private get iconNameProp(): string {
        if (this.iconName) {
            return this.iconName;
        } else {
            switch (this.state) {
                case MMessageState.Confirmation:
                    return'm-svg__confirmation';
                case MMessageState.Information:
                    return'm-svg__information';
                case MMessageState.Warning:
                    return 'm-svg__warning';
                default:
                    return'm-svg__error';
            }
        }
    }
}

const MessagePagePlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug(MESSAGE_PAGE_NAME, 'plugin.install');
        v.use(I18nPlugin);
        v.use(AccordionPlugin);
        v.use(LinkPlugin);
        v.component(MESSAGE_PAGE_NAME, MMessagePage);
    }
};

export default MessagePagePlugin;
