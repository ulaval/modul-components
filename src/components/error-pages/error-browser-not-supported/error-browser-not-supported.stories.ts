import { array, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../../conf/storybook/utils';
import { ERROR_BROWSER_NOT_SUPPORTED_NAME } from '../../component-names';
import { Link } from '../../message-page/message-page';
import ErrorBrowserNotSupported from './error-browser-not-supported';

Vue.use(ErrorBrowserNotSupported);



storiesOf(`${componentsHierarchyRootSeparator}${ERROR_BROWSER_NOT_SUPPORTED_NAME}`, module)

    .add('default', () => ({
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-browser-not-supported></m-error-browser-not-supported>
                   </div>`
    }))
    .add('title', () => ({
        props: {
            title: {
                default: text('Title', 'A Custom Title')
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-browser-not-supported :title="title"></m-error-browser-not-supported>
                   </div>`
    }))
    .add('linksDesktop', () => ({
        props: {
            linksDesktop: {
                default: array('Links[Array]', [new Link('The first custom link', 'http://www.ulaval.ca', true), new Link(`The second custom
                link`, 'http://www.google.com', true)])
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-browser-not-supported :linksDesktop="linksDesktop"></m-error-browser-not-supported>
                   </div>`
    }))
    .add('linksMobile', () => ({
        props: {
            linksMobile: {
                default: array('Links[Array]', [new Link('The first custom link', 'http://www.ulaval.ca', true), new Link(`The second custom
                link`, 'http://www.google.com', true)])
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-browser-not-supported :linksMobile="linksMobile"></m-error-browser-not-supported>
                   </div>`
    }))
    .add('hintsDesktop', () => ({
        props: {
            hintsDesktop: {
                default: array('Hints[Array]', ['My only custom hint', `My second (long) custom hint.Lorem ipsum dolor
                sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. `])
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-browser-not-supported :hintsDesktop="hintsDesktop"></m-error-browser-not-supported>
                   </div>`
    }))
    .add('hintsMobile', () => ({
        props: {
            hintsMobile: {
                default: array('Hints[Array]', ['My only custom hint', `My second (long) custom hint.Lorem ipsum dolor
                sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. `])
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-browser-not-supported :hintsMobile="hintsMobile"></m-error-browser-not-supported>
                   </div>`
    }));
