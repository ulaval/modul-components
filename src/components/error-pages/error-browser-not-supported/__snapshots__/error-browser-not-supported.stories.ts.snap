// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|m-error-browser-not-supported default 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Ce navigateur
        <br />
        n'est pas compatible
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        Une mise à niveau de votre navigateur est nécessaire pour accéder à ce site. Pour ce faire, consulter la boutique de téléchargement d'application de votre appareil
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;

exports[`Storyshots components|m-error-browser-not-supported hintsDesktop 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Ce navigateur
        <br />
        n'est pas compatible
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        Une mise à niveau de votre navigateur est nécessaire pour accéder à ce site. Pour ce faire, consulter la boutique de téléchargement d'application de votre appareil
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;

exports[`Storyshots components|m-error-browser-not-supported hintsMobile 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Ce navigateur
        <br />
        n'est pas compatible
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        My only custom hint
      </p>
      <p
        class="m-message-page__hints m--no-content"
      >
        My second (long) custom hint.Lorem ipsum dolor
                sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. 
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;

exports[`Storyshots components|m-error-browser-not-supported linksDesktop 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Ce navigateur
        <br />
        n'est pas compatible
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        Une mise à niveau de votre navigateur est nécessaire pour accéder à ce site. Pour ce faire, consulter la boutique de téléchargement d'application de votre appareil
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;

exports[`Storyshots components|m-error-browser-not-supported linksMobile 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        Ce navigateur
        <br />
        n'est pas compatible
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints"
      >
        Une mise à niveau de votre navigateur est nécessaire pour accéder à ce site. Pour ce faire, consulter la boutique de téléchargement d'application de votre appareil
      </p>
       
      <!---->
       
      <ul
        class="m-message-page__links"
      >
        <li>
          <a
            class="m-link m-message-page__link m--has-left-icon m--has-content"
            href="http://www.ulaval.ca"
            tabindex="0"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-link__icon m--has-large-stroke"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <span
              class="m-link__text"
            >
              The first custom link
            </span>
             
            <span
              class="m-i18n m-link__hidden"
            >
              Cet hyperlien s'ouvrira dans un nouvel onglet.
            </span>
          </a>
        </li>
        <li>
          <a
            class="m-link m-message-page__link m--has-left-icon m--has-content"
            href="http://www.google.com"
            tabindex="0"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-link__icon m--has-large-stroke"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <span
              class="m-link__text"
            >
              The second custom
                link
            </span>
             
            <span
              class="m-i18n m-link__hidden"
            >
              Cet hyperlien s'ouvrira dans un nouvel onglet.
            </span>
          </a>
        </li>
      </ul>
    </div>
  </article>
</div>
`;

exports[`Storyshots components|m-error-browser-not-supported title 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        A Custom Title
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        Une mise à niveau de votre navigateur est nécessaire pour accéder à ce site. Pour ce faire, consulter la boutique de téléchargement d'application de votre appareil
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;
